pydantic_ai_slim/pydantic_ai/models/gemini.py

```diff
@@ -382,6 +406,20 @@ class _GeminiRequest(TypedDict):
     Developer generated system instructions, see
     <https://ai.google.dev/gemini-api/docs/system-instructions?lang=rest>
     """
+    generation_config: NotRequired[_GeminiGenerationConfig]
+
+
+class _GeminiGenerationConfig(TypedDict, total=False):
+    """Schema for an API request to the Gemini API.
+
+    Note there are many additional fields available that have not been added yet.
+
+    See <https://ai.google.dev/api/generate-content#generationconfig> for API docs.
+    """
+
+    max_output_tokens: int
```

Comment:
In all the gemini docs it seems that they use camelCase, but in all our stuff we are using snake_case. I'm assuming both work or we would have noticed this sooner, but I didn't check that snake case works. It's probably a good idea to explicitly confirm.

Reply:
I confirmed that this is working, though we don't handle errors very gracefully, ex if `finish_reason` is not `STOP`

Reply:
```py
finish_reason: NotRequired[Annotated[Literal['STOP'], Field(alias='finishReason')]]
    """
    See <https://ai.google.dev/api/generate-content#FinishReason>, lots of other values are possible,
    but let's wait until we see them and know what they mean to add them here.
    """
```

Makes sense

Reply:
Update, I've added `MAX_TOKENS` to this literal, and things now work well:

```py
from pydantic_ai import Agent

agent = Agent(  
    'gemini-1.5-flash',
    system_prompt='Be concise, reply with one sentence.',  
)

result = agent.run_sync('Where does "hello world" come from?', model_settings={'max_tokens': 10})  
print(result.data)
"""
"Hello, world!" originated as a sample program
"""
print(result.cost())
"""
"Hello, world!" originated as a sample program
Cost(request_tokens=19, response_tokens=10, total_tokens=29, details=None)
"""
```